src/modules/settlements.ts

```diff
@@ -14,28 +14,27 @@ export namespace settlements {
 
         let settlement = Settlement.load(settlementId)
 
-        // both xdai and weth have the same amount of decimals
-        // commented code is a WIP will be addressed with:
-        // https://github.com/cowprotocol/subgraph/issues/18
         let DEFAULT_DECIMALS =  BigInt.fromI32(18)
-        let gasPriceUsd = ZERO_BD
+        let txCostUsd = ZERO_BD
+        let txCostNative = ZERO_BD
         if (network == 'xdai') {
-            // txGasPrice is in xdai
-            gasPriceUsd = convertTokenToDecimal(txGasPrice, DEFAULT_DECIMALS)
+            txCostUsd = convertTokenToDecimal(txGasPrice, DEFAULT_DECIMALS)
```

Comment:
`convertTokenToDecimal(txGasPrice, DEFAULT_DECIMALS)` is done both in the IF/ELSE should u refactor?

Reply:
Great! solved!